operator/controllers/loki/lokistack_controller.go

```diff
@@ -96,7 +96,11 @@ var (
 	})
 	createUpdateOrDeletePred = builder.WithPredicates(predicate.Funcs{
 		UpdateFunc: func(e event.UpdateEvent) bool {
-			return (e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()) ||
+			if e.ObjectOld.GetGeneration() == 0 && len(e.ObjectOld.GetAnnotations()) == 0 {
+				return true
+			}
```

Comment:
Why is that case needed and not captured below?

Reply:
Because some objects, like the storage secret, don't have these attributes. In that case, there is no point int checking for changes in those attributes.

Reply:
Let me ask again, why don't they have such attributes? Or at least the generation?

Reply:
Ah the keyword `replicaset` was the ringing bell for me. Thank you :)

OTH this means that this function does not track if any real updates happen on the secret. Maybe checking the `resourceVersion` is a good alternative here?

Reply:
I believe it does. The only time an Update event occurs is when a change occurs on the object, and that results in `resourceVersion` changing as well. Comparing `resourceVersion` would be redundant in this case IMO, but it shouldn't hurt to have it.